test/java/seedu/addressbook/commands/ViewCommandTest.java

```diff
@@ -0,0 +1,154 @@
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.tag.UniqueTagList;
+import seedu.addressbook.util.TestUtil;
+import seedu.addressbook.util.TypicalPersons;
+
+public class ViewCommandTest {
+    private TypicalPersons td = new TypicalPersons();
+
+    private AddressBook typicalAddressBook = td.getTypicalAddressBook();
+    private AddressBook emptyAddressBook = TestUtil.createAddressBook();
+    private List<ReadOnlyPerson> emptyPersonList = Collections.emptyList();
+    private List<ReadOnlyPerson> listWithAllTypicalPersons = Arrays.asList(td.getTypicalPersons());
+    private List<ReadOnlyPerson> listWithSomeTypicalPersons = Arrays.asList(td.amy, td.candy, td.dan);
+
+    @Test
+    public void execute_invalidIndex_returnsInvalidIndexMessage() {
+        // empty addressbook
+        assertViewErrorInvalidIndex(emptyAddressBook, emptyPersonList, 1);
+
+        // non-empty addressbook
+        assertViewErrorInvalidIndex(typicalAddressBook, listWithAllTypicalPersons, -1);
+        assertViewErrorInvalidIndex(typicalAddressBook, listWithAllTypicalPersons, 0);
+        assertViewErrorInvalidIndex(typicalAddressBook, listWithAllTypicalPersons,
+                listWithAllTypicalPersons.size() + 1);
+    }
+
+    @Test
+    public void execute_personNotInAddressBook_returnsPersonNotInAddressBookMessage() throws Exception {
+        // generate list with person not in addressbook, add to list
+        ReadOnlyPerson someone = new Person(new Name("me"),
+                                            new Phone("123", true),
+                                            new Email("some@hey.go", true),
+                                            new Address("nus", false),
+                                            new UniqueTagList(Collections.emptySet()));
+        List<ReadOnlyPerson> listWithExtraPerson
+                = new ArrayList<ReadOnlyPerson>(listWithAllTypicalPersons);
+        listWithExtraPerson.add(someone);
+
+        // empty addressbook
+        assertViewErrorPersonNotInAddressBook(emptyAddressBook, listWithExtraPerson, 1);
+
+        // non-empty addressbook
+        assertViewErrorPersonNotInAddressBook(typicalAddressBook, listWithExtraPerson,
+                listWithExtraPerson.size());
+    }
+
+    @Test
+    public void execute_validIndex_returnsPersonDetails() {
+        // person with no private information
+        assertViewSuccess(typicalAddressBook, listWithAllTypicalPersons, 1);
+
+        // person with some private information
+        assertViewSuccess(typicalAddressBook, listWithAllTypicalPersons, 2);
+
+        // person with all private information
+        assertViewSuccess(typicalAddressBook, listWithAllTypicalPersons, 4);
+
+        // addressbook has more people than the list
+        // This can happen when view from filtered list caused by some commands(eg. FindCommand)
```

Comment:
This sentence is not very clear.
`This can happen when a command causes the list to show only a sub-set of persons (e.g. FindCommand).`

Also, full sentences should end with full stop.